### PR TITLE
Fixing Cactuses

### PR DESCRIPTION
**Pull request by TartarusKelvin**
#### Summary
`SUMMARY: Content "Added cactus pads so that cacti can be picked"`

#### Purpose of change
Fixes#29955 - Previously picking a cactus would act the same as picking a poppy

#### Describe the solution
I add examining a cactus as a function and created cactus pad a form of edible cactus

#### Additional context
Maybe in future you should be able to get water from cacti with the appropriate skill

### CODE REVIEW

**Comment by Zireael07** [comment]
What are feather and gravedigging commits doing in this PR? Forked off a wrong branch?

**Comment by TartarusKelvin** [comment]
Yeah I believe so

File: src/iexamine.cpp [hunk 1]
```diff
@@ -1555,6 +1555,40 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
     }
 }
 
+/**
+ * Prompt pick (or drink nectar if able) poppy bud. Not safe for player.
+ *
+ * Drinking causes: -25 hunger, +20 fatigue, pkill2-70 effect and, 1 in 20 pkiller-1 addiction.
+ * Picking w/ env_resist < 5 causes 1 in 3  sleep for 12 min and 4 dmg to each leg
+ */
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: Change comments to reflect the actual info on cactus pads, or remove them altogether.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -1555,6 +1555,40 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
...
+        return;
+    }
+
+    if( !query_yn( _( "Pick %s?" ), g->m.furnname( examp ) ) ) {
+        none( p, examp );
+        return;
+    }
+
+    if( one_in( 6 ) ) {
+        add_msg( m_bad, _( "The cactus's nettles sting you!" ) );
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]:
```suggestion
        add_msg( m_bad, _( "The cactus' nettles sting you!" ) );
```

File: src/iexamine.cpp [hunk 3]
```diff
@@ -1555,6 +1555,40 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
...
+        p.apply_damage( nullptr, bp_arm_l, 4 );
+        p.apply_damage( nullptr, bp_arm_r, 4 );
+    }
+
+    g->m.furn_set( examp, f_null );
+
+    if( p.can_pickWeight( item( "cactus_pad" ), true ) &&
+        p.can_pickVolume( item( "cactus_pad" ), true ) ) {
+        p.i_add( item( "cactus_pad" ) );
+        p.add_msg_if_player( _( "You harvest: cactus pad" ) );
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]:
```suggestion
        p.add_msg_if_player( _( "You harvest: cactus pad." ) );
```

File: src/iexamine.cpp [hunk 4]
```diff
@@ -1555,6 +1555,40 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
...
+    } else {
+        g->m.add_item_or_charges( p.pos(), item( "cactus_pad" ) );
+        p.add_msg_if_player( _( "You harvest and drop: cactus pad" ) );
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]:
```suggestion
        p.add_msg_if_player( _( "You harvest and drop: cactus pad." ) );
```

File: data/json/items/comestibles/raw_fruit.json [hunk 5]
```diff
@@ -602,5 +602,24 @@
     "flags": [ "SMOKABLE" ],
     "smoking_result": "dry_fruit",
     "vitamins": [ [ "vitA", 17 ], [ "vitC", 18 ], [ "calcium", 2 ], [ "iron", 4 ] ]
+  },
+  {
+	  "type": "COMESTIBLE",
+	  "id": "cactus_pad",
+	  "name": "cactus pad",
+	  "name": "cactus pads",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/raw_fruit.json]: Tabs.

File: data/json/items/comestibles/raw_fruit.json [hunk 6]
```diff
@@ -602,5 +602,24 @@
...
+    "weight": 100,
+    "color": "green",
+    "spoils_in": "1 days",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "quench": 15,
+    "healthy": 1,
+    "calories": 100,
+    "description": "A edible pad of a cactus",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/raw_fruit.json]:
```suggestion
    "description": "A edible pad of a cactus.",
```

File: src/iexamine.cpp [hunk 7]
```diff
@@ -1555,6 +1555,37 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
...
+    }
+
+    g->m.furn_set( examp, f_null );
+
+    if( p.can_pickWeight( item( "cactus_pad" ), true ) &&
+        p.can_pickVolume( item( "cactus_pad" ), true ) ) {
+        p.i_add( item( "cactus_pad" ) );
+        p.add_msg_if_player( _( "You harvest: cactus pad." ) );
+    } else {
+        g->m.add_item_or_charges( p.pos(), item( "cactus_pad" ) );
```
Comment (codemime) [inline src/iexamine.cpp]: I wouldn't rely on optimizations here: please extract a temp to avoid re-creating the `item`.

File: src/iexamine.cpp [hunk 8]
```diff
@@ -5177,6 +5208,7 @@ iexamine_function iexamine_function_from_string( const std::string &function_nam
             { "door_peephole", &iexamine::door_peephole },
             { "fswitch", &iexamine::fswitch },
             { "flower_poppy", &iexamine::flower_poppy },
+	    { "flower_cactus", &iexamine::flower_cactus },
```
Comment (codemime) [inline src/iexamine.cpp]: There's a tab character in the line.

File: src/iexamine.cpp [hunk 9]
```diff
@@ -1555,6 +1555,38 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
...
+
+    if( one_in( 6 ) ) {
+        add_msg( m_bad, _( "The cactus' nettles sting you!" ) );
+        p.apply_damage( nullptr, bp_arm_l, 4 );
+        p.apply_damage( nullptr, bp_arm_r, 4 );
+    }
+
+    g->m.furn_set( examp, f_null );
+
+    item tmpCactus = item( "cactus_pad" );
```
Comment (codemime) [inline src/iexamine.cpp]: Snake_case please: it's `tmp_cactus` (or just `cactus` which is even better).

File: src/iexamine.cpp [hunk 10]
```diff
@@ -1555,6 +1555,38 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
...
+    if( p.can_pickWeight( item( "cactus_pad" ), true ) &&
+        p.can_pickVolume( item( "cactus_pad" ), true ) ) {
```
Comment (codemime) [inline src/iexamine.cpp]: These are still duplications.

File: src/iexamine.cpp [hunk 11]
```diff
@@ -1555,6 +1555,38 @@ void iexamine::flower_poppy( player &p, const tripoint &examp )
...
+        p.i_add( tmpCactus );
+        p.add_msg_if_player( _( "You harvest: cactus pad." ) );
```
Comment (codemime) [inline src/iexamine.cpp]: ```p.add_msg_if_player( _( "You harvest: %s." ), cactus_pad.tname() );```

File: src/iexamine.cpp (hunk 10 above)
Comment (TartarusKelvin) [reply src/iexamine.cpp]: whoops must have missed those

**Comment by codemime** [comment]
Jenkins rebuild.

**Comment by AMurkin** [comment]
Right and easy way to format json files - use http://dev.narc.ro/cataclysm/format.html.

File: data/json/items/comestibles/veggy_dishes.json [hunk 12]
```diff
@@ -696,5 +696,23 @@
...
+    "id": "nopalitos",
+    "name": "nopalito",
+    "weight": 200,
+    "color": "green",
+    "spoils_in": "10 days",
+    "symbol": "%",
+    "quench": 30,
+    "healthy": 1,
+    "calories": 55,
+    "description": "A less prickly version of cactus pads",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/veggy_dishes.json]:
```suggestion
    "description": "A less prickly version of cactus pads.",
```

**Pull request merged**